torch/_dynamo/symbolic_convert.py

```diff
@@ -341,6 +342,23 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
                 self.jump(inst)
                 return
 
+            if isinstance(value, SymNodeVariable):
+                # if the assertion is normal shape expression.
+                # just install guard and bail out.
+                try:
+                    result = value.evaluate_expr()
+                    if not result:
+                        raise RuntimeError(
+                            """Symbolic expression for assertion is evaluated to False."""
+                            """ Did you make sure the eager version succeed?"""
+                        )
```

Comment:
This still is not right. The RuntimeError will result in an InternalDynamoException, but that's inappropriate. unsupported() is a better choice here

Reply:
Why do we want to graph break? This means the eager code is already wrong right? Are you saying we should let it fail in eager by switching to python?